### PR TITLE
feat: init parallel image modal

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes FW-663 FW-675

## Checklist

- [ ] I have tested these changes locally.
- [ ] I have reviewed the code changes.
- [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ interface ModalProps {
 
 export function Modal({ backdrop = true, open, onClose, children, className }: ModalProps) {
     return (
-        <Transition appear show={open} as={Fragment}>
+        <Transition appear show={open} as="div" className="test">
```
Comment (guanbinrui) [inline src/components/Modal.tsx:18]:
```suggestion
        <Transition appear show={open} as="div">
```

File: src/components/Posts/Attachment.tsx [hunk 2]
```diff
@@ -108,28 +108,25 @@ export const Attachments = memo<AttachmentsProps>(function Attachments({ attachm
...
-                            PreviewImageModalRef.open({
-                                images: [asset.uri],
-                                current: asset.uri,
-                                post,
-                            });
-                        }}
-                        src={formatImageUrl(asset.uri, ATTACHMENT)}
-                        alt={formatImageUrl(asset.uri, ATTACHMENT)}
-                    />
+                    <Link href={getPostImageUrl(post, 1)} scroll={false}>
```
Comment (guanbinrui) [inline src/components/Posts/Attachment.tsx:111]: It starts from 1?

File: src/app/globals.css [hunk 3]
```diff
@@ -260,3 +260,49 @@ body {
...
+
+.preview-actions .slick-next:before {
+    display: none;
+}
+
+.preview-actions .slick-disabled {
+    display: none !important;
+}
+
+@media screen and (max-width: 980px) {
```
Comment (guanbinrui) [inline src/app/globals.css:290]: `980` not a configured breadkpoint. 

image

File: src/app/@modal/(.)post/[id]/photos/[index]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+interface Props {
+    params: {
+        id: string;
+        index: string;
+    };
+    searchParams: { source: SourceInURL };
+}
+
+interface CustomArrowProps {
```
Comment (guanbinrui) [inline src/app/@modal/(.)post/[id]/photos/[index]/page.tsx:32]:
```suggestion
interface CustomArrowProps extends ClickableButtonProps {
```

File: src/components/Posts/Attachment.tsx (hunk 2 above)
Comment (nuanyang233) [reply src/components/Posts/Attachment.tsx:111]: I want to make the url more understandable, `/photos/0` looks a bit strange.

**Review by guanbinrui — APPROVED** [review]

File: src/components/VirtualList/index.tsx [hunk 5]
```diff
@@ -20,7 +22,8 @@ export function VirtualList<ItemData = unknown, Context = unknown>({
         if (!listKey) return;
         const index = scrollIndex[listKey];
         if (!index) return;
-        ref.current?.scrollToIndex(index);
+        // index - 2 is to keep the target as centered as possible.
```
Comment (swkatmask) [inline src/components/VirtualList/index.tsx:25]: You might use `scrollIntoView({ index, align: 'center' })`

**Pull request merged**